clients.go

```diff
@@ -93,11 +93,13 @@ func NewClients(config ClientsConfig) (*Clients, error) {
 
 	var ctrlClient client.Client
 	{
-		if config.AddToScheme != nil {
+		if config.SchemeBuilder != nil {
 			// Extend the global client-go scheme which is used by all the tools under
 			// the hood. The scheme is required for the controller-runtime controller to
 			// be able to watch for runtime objects of a certain type.
-			err = config.AddToScheme(scheme.Scheme)
+			schemeBuilder := runtime.SchemeBuilder(config.SchemeBuilder)
```

Comment:
Slightly longer than in https://github.com/giantswarm/k8sclient/pull/23#discussion_r353333285 due to compile issues but I think it's still readable and I prefer the updated interface.